include/deal.II/base/data_out_base.h

```diff
@@ -3030,6 +3030,55 @@ class DataOutInterface
   write_pvtu_record(std::ostream &                  out,
                     const std::vector<std::string> &piece_names) const;
 
+  /**
+   * This function combines DataOutInterface::write_vtu(), or
+   * DataOutInterface::write_vtu_in_parallel(), with
+   * DataOutInterface::write_pvtu_record(). It automatically constructs
+   * the filename of the generated output files and writes a pvtu file if
+   * necessary. A specified @p folder and a @p filename_without_extension
+   * form the first part of the filename. The filename is then extended with
+   * a @p counter labeling the current timestep/iteration/etc., the processor ID,
+   * and finally the .vtu/.pvtu ending. Since the number of timesteps to be
+   * written depends on the application, the number of digits to be reserved in
+   * the filename can be specified as parameter @p n_digits_counter, and the number
+   * is not padded with leading zeros if this parameter is left at its default
+   * value numbers::invalid_unsigned_int. If more than one file identifier
+   * is needed (e.g. time step number and iteration counter of solver), the
+   * last identifier is used as @p counter, while all other identifiers have to be
+   * added to @p filename_without_extension when calling this function. In a
+   * parallel setting, several files are typically written per time step. The
+   * number of files written in parallel depends on the number of MPI processes
+   * (see parameter @p mpi_communicator with default value MPI_COMM_WORLD), and a
+   * specified number of @p n_groups with default value 0. The background is that
+   * VTU file output supports grouping files from several CPUs into a given
+   * number of files using MPI I/O when writing on a parallel filesystem. The
+   * default value of @p n_groups is 0, meaning that every MPI rank will write one
+   * file. A value of 1 will generate one big file containing the solution over
+   * the whole domain, while a larger value will create @p n_groups files (but not
+   * more than there are MPI ranks). A.pvtu file is written if more than one
```

Comment:
I would always write the .pvtu file based on the name of this function. We typically want to generate a master record with all .pvtu files as well, which would be awkward if n_ranks==1 is a special case.

Reply:
done in https://github.com/dealii/dealii/pull/8904/commits/f07a8bf9bce24520305b079bc179599d78782753